### PR TITLE
feat: initial configuration

### PR DESCRIPTION
**Pull request by njfamirm**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+  "name": "one",
```
Comment (AliMD) [inline package.json:2]: @one/...

Comment (AliMD) [reply package.json:2]: Turn it private

**Review by AliMD — CHANGES_REQUESTED** [review]
Thanks

**Comment by AliMD** [comment]
@asmanp2012 please review

File: package.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "@one",
```
Comment (asmanp2012) [inline package.json:2]: Rename to "@one/core"

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+// Reference: https://www.typescriptlang.org/tsconfig
+
+{
+  "compilerOptions": {
+    "strict": true,
+    "lib": ["es2020"],
+    "target": "ES2020",
```
Comment (asmanp2012) [inline tsconfig.json:7]: change target to 2019

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
+// Reference: https://www.typescriptlang.org/tsconfig
+
+{
+  "compilerOptions": {
+    "strict": true,
+    "lib": ["es2020"],
```
Comment (asmanp2012) [inline tsconfig.json:6]: Add this libs `ScriptHost`, `WebWorker`, `DOM`, `DOM.Iterable`

File: tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    "target": "ES2020",
+    "module": "ES2020",
+    "declaration": true,
+    "skipLibCheck": true,
```
Comment (asmanp2012) [inline tsconfig.json:10]: Add `outDir`

File: .eslintignore [hunk 6]
```diff
@@ -0,0 +1,4 @@
+# Reference: https://eslint.org/docs/user-guide/configuring/ignoring-code
+
+**/*.js
```
Comment (asmanp2012) [inline .eslintignore:3]: remove **/
and add `mjs`, `map`, `d.ts` type files

Comment (asmanp2012) [inline .eslintignore:3]: remove **/
and add `mjs`, `map`, `d.ts` type files

File: .eslintrc [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+
+{
+  "root": true,
+  "extends": [
+    "plugin:@typescript-eslint/eslint-recommended",
+    "plugin:@typescript-eslint/recommended"
+  ],
+  "parser": "@typescript-eslint/parser",
+  "plugins": ["@typescript-eslint"],
+  "parserOptions": { "project": ["./tsconfig.json"] },
```
Comment (asmanp2012) [inline .eslintrc:11]: Make a copy of tsconfig file with the `ts-lint.json` name. and put this inside the "parserOptions"

File: .eslintrc [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+  "rules": {
+    "@typescript-eslint/strict-boolean-expressions": [
```
Comment (asmanp2012) [inline .eslintrc:13]: Use `@typescript-eslint/no-inferrable-types` instead of this.

File: .eslintrc [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+      2,
+      {
+        "allowString": false,
+        "allowNumber": false
+      }
+    ]
+  },
+  "ignorePatterns": ["node_modules/"]
```
Comment (asmanp2012) [inline .eslintrc:21]: Remove this line.

**Review by asmanp2012 — CHANGES_REQUESTED** [review]
Complete the files list in ignore files and remove the extra config.

**Comment by njfamirm** [comment]
ok?

**Comment by njfamirm** [comment]
tsconfig additional commecnt was removed!

**Review by asmanp2012 — CHANGES_REQUESTED** [review]
Reject

**Pull request closed without merging**